### PR TITLE
Islandora-ize Document media type (Issue 1459)

### PR DESCRIPTION
**Pull request by seth-shaw-unlv**
**GitHub Issue**: Resolves https://github.com/Islandora/documentation/issues/1459

# What does this Pull Request do?

Adds the necessary configs to use the Document media type just like the File media type.

# What's new?
A in-depth description of the changes made by this PR. Technical details and
 possible side effects.

* Added
    * Document to 'All Media' context
    * All the Islandora Media fields to Document
    * A default, PDFjs, and source display modes for Document
    * A default form view for Document
    * An RDF mapping for Document
* Does this change require documentation to be updated? I don't think so.
* Does this change add any new dependencies? No.
* Does this change require any other modifications to be made to the repository
 (i.e. Regeneration activity, etc.)? No.
* Could this change impact execution of existing code? No.

# How should this be tested?

- Attempt to add a Document media and notice there are none of the usual Islandora fields (e.g. Media Of or Media Use).
- Apply the PR
- Import the Feature
- Attempt to add a Document media again and see all the usual fields. Creating the media will make it appear in Fedora as usual. 
- *Bonus* upload a PDF as a Document media (use can select both 'Original File' & 'Service File') and use the PDFjs display hint on the linked node to see your PDF show up in the PDFjs viewer.

# Additional Comments

Please squash when you merge. Thanks.

# Interested parties
@bseeger and @Islandora/8-x-committers

### CODE REVIEW

**Comment by dannylamb** [comment]
Travis is unhappy: https://travis-ci.com/github/Islandora/islandora_defaults/jobs/299223994#L1309

Looks like a config dependency issue.

**Comment by seth-shaw-unlv** [comment]
Moving my Slack notes here:

@dannylamb, [Travis is still using Drupal 8.6 for testing](https://travis-ci.com/github/Islandora/islandora_defaults/jobs/299223994#L443), which is why this failed. Two options: we bump travis up to 8.8.x **or** we move the configs for the Document media type into `/config/optional/` . Thoughts?

The [travis config simply says 8.x](https://github.com/Islandora/islandora_defaults/blob/8.x-1.x/.travis.yml#L17), so I wonder why Travis is using such an old version....

File: .travis.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ services:
 
 branches:
   only:
-    - /^8.x/
+    - /^8.8.x/
```
Comment (dannylamb) [inline .travis.yml]: Revert this back to `^8.x`

Comment (seth-shaw-unlv) [reply .travis.yml]: 👍

**Comment by dannylamb** [comment]
After testing both this and the PR on islandora, I'm getting no thumbnails and an error when loading the document in pdfjs

image

Am I missing something?

**Comment by dannylamb** [comment]
Those fixes work. Pulled in the changes and re-imported the feature :+1:

**Pull request merged**